### PR TITLE
Feature/view discover

### PR DESCRIPTION
**Pull request by isaacmartinezimv**
[[FARM-68,](https://amidodevelopment.atlassian.net/secure/RapidBoard.jspa?rapidView=150&projectKey=FARM&view=planning&selectedIssue=FARM-68) [FARM-86](https://amidodevelopment.atlassian.net/secure/RapidBoard.jspa?rapidView=150&projectKey=FARM&view=planning&selectedIssue=FARM-86)]

#### 📲 What

Get list of articles from firestore
Show list of articles
Article list item design
Navigate to article description
Show image and enriched text

#### 🤔 Why

Is important for the users to see a list of articles and to navigate them to read it.
		
#### 🛠 How

Through a list view of cards with rounded borders, with expanded widget for texts.

#### 👀 See

gif_graphic12

#### ✅ Acceptance criteria

- [X] Manually tested
- [X] Showcase video / automation test
- [X] Passing all tests
- [ ] Rebased/merged with latest changes from dev/release and re-tested
- [X] Removed TODOs and FIXMEs

#### ⏰ TODO

- [ ] Change the icon to navigate article description

#### 🕵️‍♂️ How to test

Navigate to discover tab, and go into an article description

#### Reviewers

@farmsmart/amido @farmsmart/wamf @emilianowearemobile @maribel4amido

### CODE REVIEW

File: lib/ui/discover/discover_child_item.dart
Comment (emilianowearemobile) [inline lib/ui/discover/discover_child_item.dart]: Check Icons.arrow_forward_ios for the arrow asset :)

**Comment by LeeMatthewHiggins** [comment]
Don't forget to add reviewers to all PR´s. Min of 2 people should approve before you merge. We can use https://gifox.io. to create nice animated Demos of work.

**Comment by LeeMatthewHiggins** [comment]
Also let´s add the Jira links so they are clickable.

File: lib/data/managers/firestore_manager.dart
Comment (emilianowearemobile) [inline lib/data/managers/firestore_manager.dart:60]: Maybe we can do a method that returns this query, with schema as parameter.
Like prepareFirestoreQuery(schema.ARTICLE)
and prepareFirestoreQuery(schema.CROPS)
so it can be reused :)

File: lib/data/repositories/discover_repository.dart
Comment (emilianowearemobile) [inline lib/data/repositories/discover_repository.dart]: Repository should reference to what data provides...
in this case, it provides Articles... so it should be an ArticlesRepository... 
I suggest that renaming.

File: lib/redux/home/home_state.dart
Comment (emilianowearemobile) [inline lib/redux/home/home_state.dart]: This will likely generate merge conflicts as i deleted it on my branch.
Doesnt matters but, we need to keep in mind that this line should disappear :)

File: lib/redux/middleware/discover_middleware.dart
Comment (emilianowearemobile) [inline lib/redux/middleware/discover_middleware.dart:6]: This makes sense to refer it as Discover and not ArticleMiddleware :)

Comment (emilianowearemobile) [inline lib/redux/middleware/discover_middleware.dart:6]: +1

File: lib/redux/middleware/navigation_middleware.dart
Comment (emilianowearemobile) [inline lib/redux/middleware/navigation_middleware.dart:18]: maybe a switch case is better for this class?

File: lib/ui/app_bar.dart
Comment (emilianowearemobile) [inline lib/ui/app_bar.dart:21]: Probably, we wont want to have access to the profile from this screen.
But we can ask.

File: lib/ui/discover/discover_child_item.dart
Comment (emilianowearemobile) [inline lib/ui/discover/discover_child_item.dart]: I think we should keep, every "child" inside a private method, in the class...
this way you would have like 
Row ( children [
buildTitle,
buildImage,
buildSubtitle
]

all as private members of the class.
I think its a healthy practice, agree?

**Review by emilianowearemobile — CHANGES_REQUESTED** [review]
A bunch of suggestions :)

**Comment by harrylusk** [comment]
I've spoken to Emma and it's our preference that the designs be aligned to the UX as per the acceptance criteria in the ticket. Is this possible? Does this require significant rework?

Screenshot 2019-05-07 at 14 10 10

**Review by maribel4amido — COMMENTED** [review]
When specifying the JIRA, please ensure that the Sprint JIRA is specified (and not just the subtasks). 

This allows our product owners/ba to easily associate a PR to a user story.

**Review by harrylusk — CHANGES_REQUESTED** [review]
I've spoken to Emma and it's our preference that the designs be aligned to the UX as per the acceptance criteria in the ticket. Is this possible? Does this require significant rework?

Screenshot 2019-05-07 at 14 10 10

File: lib/data/managers/firestore_manager.dart
Comment (maribel4amido) [reply lib/data/managers/firestore_manager.dart]: In addition, there are other filters such as the environment and locale that will need to be considered.

File: lib/ui/discover/discover_child_item.dart
Comment (isaacmartinezimv) [reply lib/ui/discover/discover_child_item.dart]: Done! thnks ;)

File: lib/data/repositories/discover_repository.dart
Comment (isaacmartinezimv) [reply lib/data/repositories/discover_repository.dart]: Done!

File: lib/ui/discover/discover_child.dart
Comment (maribel4amido) [inline lib/ui/discover/discover_child.dart]: Add todo as per https://amidodevelopment.atlassian.net/browse/FARM-203

File: lib/ui/discover/discover_child.dart
Comment (maribel4amido) [inline lib/ui/discover/discover_child.dart]: I suggest placing padding styles in a common file like the TextStyles.

File: lib/ui/discover/discover_child_item.dart
Comment (maribel4amido) [inline lib/ui/discover/discover_child_item.dart]: Place string constants in a common file

File: lib/ui/discover/discover_child_item.dart
Comment (maribel4amido) [inline lib/ui/discover/discover_child_item.dart]: Please refer to the colors common file or colors class

File: lib/ui/discover/discover_child_item.dart
Comment (maribel4amido) [inline lib/ui/discover/discover_child_item.dart]: Please clean unused comments

File: lib/ui/discover/discover_detail_screen.dart
Comment (maribel4amido) [inline lib/ui/discover/discover_detail_screen.dart]: Should decorations be defined same as TextStyle?

File: lib/ui/discover/discover_child.dart
Comment (maribel4amido) [inline lib/ui/discover/discover_child.dart]: I don't quite get the name HomeDiscoverChild.  It looks to be the view articles page but I got confused on the use of Home and Child.

Comment (maribel4amido) [reply lib/ui/discover/discover_child.dart]: Seems to be the same approach as myplot.

From what I understand: 
child -> same concept as page
detail screen -> the actual detail screen of the child

In my understanding child and detail can be the same thing but it's different here. The child is actually the parent screen.
Is my understanding correct?

**Review by maribel4amido — CHANGES_REQUESTED** [review]

File: lib/ui/app_bar.dart
Comment (isaacmartinezimv) [reply lib/ui/app_bar.dart:21]: i put it because is in the UX/UI designs

File: lib/redux/middleware/navigation_middleware.dart
Comment (isaacmartinezimv) [reply lib/redux/middleware/navigation_middleware.dart:18]: with switch isn't working for the moment. The middleware seems do not support switch cases yet.

File: lib/data/managers/firestore_manager.dart
Comment (isaacmartinezimv) [reply lib/data/managers/firestore_manager.dart:60]: Done! Thnks for the help

File: lib/data/model/article_entity.dart
Comment (albertmiro) [inline lib/data/model/article_entity.dart]: Can we move the string literals to constants?

**Comment by Thaiswamf** [comment]
> I've spoken to Emma and it's our preference that the designs be aligned to the UX as per the acceptance criteria in the ticket. Is this possible? Does this require significant rework?
> 
> Screenshot 2019-05-07 at 14 10 10



> I've spoken to Emma and it's our preference that the designs be aligned to the UX as per the acceptance criteria in the ticket. Is this possible? Does this require significant rework?
> 
> Screenshot 2019-05-07 at 14 10 10

Hi @harrylusk, yes, we can change the design removing the rounded corners and the shadow, but the changes to add the line will not be as easy. 
We have two options to do the changes to add the line: Option 1, we can add a line under each article that fill the entire screen (with the same device width) easily on this PR. Option 2, if we want to put the separating line only above the text and we want that it has a width smaller than the size of the screen, it would require some time because it will be needed to change the development of some components on the screen and these tasks would be needed to be defined as a new task for the next sprint.
What option do you prefer to follow for this screen? Thank you

**Comment by jessicahernandezhv** [comment]
Hello @harrylusk, yesterday we've been investigating about the Option 2 that @Thaiswamf proposed (make a separator only above the text, as the UI design shows).

After a deep research, we just found an option to make it, changing the development of the minimum number of components, that make possible to include it in this PR. Thank you.

**Comment by harrylusk** [comment]
@jessicahernandezhv @Thaiswamf thanks for your time researching this. If option 2 is simple, let's go with that. Otherwise, I'm happy if the line displays across the full width of the screen.

**Comment by harrylusk** [comment]
@jessicahernandezhv also, can we change the icon from the coins to the arrow?

File: lib/ui/app_bar.dart
Comment (albertmiro) [inline lib/ui/app_bar.dart]: If we are not going to use it, let's remove the comment

File: lib/ui/discover/discover_child.dart
Comment (albertmiro) [reply lib/ui/discover/discover_child.dart]: I am agree with you @maribel4amido, let's find a better name if this one is making confusion

File: lib/ui/discover/discover_child.dart
Comment (albertmiro) [inline lib/ui/discover/discover_child.dart]: ` return Text("Error");`
Let's move this string to a const value, or some place where we can localize later on.

File: lib/ui/discover/discover_child_item.dart
Comment (albertmiro) [inline lib/ui/discover/discover_child_item.dart]: Move the string to a common strings file, we shouldn't have hardcoded values in the app

File: lib/ui/discover/discover_child_item.dart
Comment (isaacmartinezimv) [reply lib/ui/discover/discover_child_item.dart]: that color property has been deleted to approach the view to the deign pattern, but yes it should be refer to our colors common file, Thnks !!

File: lib/ui/discover/discover_child_item.dart
Comment (isaacmartinezimv) [reply lib/ui/discover/discover_child_item.dart]: Done!

File: lib/ui/discover/discover_viewmodel.dart
Comment (albertmiro) [inline lib/ui/discover/discover_viewmodel.dart]: Maybe we can add the `store.dispatch(GoToArticleDetailAction(article))` to a new line to follow the same as the next line with the fetchArticles

File: lib/ui/discover/discover_viewmodel.dart
Comment (albertmiro) [inline lib/ui/discover/discover_viewmodel.dart]: `new` is no longer needed in Dart 2

File: lib/ui/discover/discover_detail_screen.dart
Comment (isaacmartinezimv) [reply lib/ui/discover/discover_detail_screen.dart]: that decoration was no necessary, so i just deleted.
We are in process of making every "child" inside a private method, in the class that uses it, so if after that is still necessary to make a common BoxDecorations file, it's no problem at all :)

File: lib/ui/discover/discover_detail_screen.dart
Comment (albertmiro) [inline lib/ui/discover/discover_detail_screen.dart:51]: Can we move the Widget to a method that can return it, and then another one to return the Padding?

We would have something like, I think is more readable:
```
  child: ListView(children: <Widget>[
                getFadeImage(selectedArticleData),
                getPadding(selectedArticleData),
              ]))
```

File: lib/ui/discover/discover_child.dart
Comment (isaacmartinezimv) [reply lib/ui/discover/discover_child.dart]: Child is referred to the bottomBar, so myPlot, P/L, Discover and Community are every child of the bottom bar.
Item is every element listed.
Detail is the screen inside every item in the list

But you are right with the naming,i think is a little bit confusing so we can rename all the children of bottom bar to Page, like:

 HomeDiscoverChild => HomeDiscoverPage,
 and HomeMyPlotChild => HomeMyPlotPage

Thnks!

**Comment by isaacmartinezimv** [comment]
> @jessicahernandezhv also, can we change the icon from the coins to the arrow?

yeah, its changed, thnks Harry !

File: lib/data/model/article_entity.dart
Comment (isaacmartinezimv) [reply lib/data/model/article_entity.dart]: for sure, Done!

File: lib/ui/app_bar.dart
Comment (isaacmartinezimv) [reply lib/ui/app_bar.dart]: Done!

File: lib/ui/discover/discover_child.dart
Comment (isaacmartinezimv) [reply lib/ui/discover/discover_child.dart]: Done!

File: lib/ui/discover/discover_child_item.dart
Comment (isaacmartinezimv) [reply lib/ui/discover/discover_child_item.dart]: Done!

File: lib/ui/discover/discover_viewmodel.dart
Comment (isaacmartinezimv) [reply lib/ui/discover/discover_viewmodel.dart]: totally agree ! it keeps the code more easy to read.

File: lib/ui/discover/discover_viewmodel.dart
Comment (isaacmartinezimv) [reply lib/ui/discover/discover_viewmodel.dart]: yes, sometimes i forgot that, it's removed now, thnks !

File: lib/ui/discover/discover_child.dart
Comment (maribel4amido) [reply lib/ui/discover/discover_child.dart]: Thanks guys!

**Review by maribel4amido — DISMISSED** [review]

File: lib/ui/discover/discover_child_item.dart
Comment (isaacmartinezimv) [reply lib/ui/discover/discover_child_item.dart]: Done !

File: lib/ui/discover/discover_detail_screen.dart
Comment (isaacmartinezimv) [reply lib/ui/discover/discover_detail_screen.dart:51]: Done !

Comment (emilianowearemobile) [reply lib/ui/discover/discover_detail_screen.dart:51]: Albert, don't copy my comments! xD

Comment (albertmiro) [reply lib/ui/discover/discover_detail_screen.dart:51]: @emilianowearemobile I haven't seen yours! <image src="https://user-images.githubusercontent.com/15966505/57441666-b0bf8a80-724a-11e9-99f0-358d240eb456.png" width=8%/>

File: lib/utils/strings.dart
Comment (emilianowearemobile) [inline lib/utils/strings.dart]: These are not "UI Strings" that are going to be localized neither shown to the user.
They are constants for firebase, check firebase_const file :)

**Review by emilianowearemobile — DISMISSED** [review]

**Pull request merged**
